driver/src/price_finding/naive_solver.rs

```diff
@@ -340,4 +340,41 @@ pub mod tests {
         let res = solver.find_prices(&orders, &state);
         assert_eq!(U256::from(0), res.unwrap().surplus);
     }
+
+    #[test]
+    fn test_e2e_example() {
```

Comment:
Why do we need this test case? Should this be a separate PR?

Reply:
I introduced it to figure out what we needed in the e2e-test and figured it made sense to leave here since it came in handy for these changes. However, I discovered only later that I only needed this unit test because I hadn't included the appropriate deposits. I then found it interesting to have an example with zero surplus.

Reply:
Would be happy to remove this if you insist, but might not bother including it in a separate PR in that case...